hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -338,18 +339,22 @@ protected void deleteInstantFile(HoodieInstant instant) {
    *
    * <p>TODO: [HUDI-6885] Depreciate HoodieActiveTimeline#getInstantFileName and fix related tests.
    */
-  protected String getInstantFileName(HoodieInstant instant) {
-    if (instant.isCompleted() && instant.getCompletionTime() == null) {
-      return getInstantsAsStream().filter(s -> s.equals(instant))
-          .findFirst().orElseThrow(() -> new HoodieIOException("Cannot find the instant" + instant))
-          .getFileName();
+  protected String getInstantFileName(HoodieInstant instant, int versionCode) {
+    if (versionCode < HoodieTableVersion.EIGHT.versionCode()) {
```

Comment:
It looks like the logic is changed for test code, maybe we should just restrict the logic in the tests either.